### PR TITLE
feat(dashboard): UI for charts dashboard page

### PR DESCRIPTION
**Pull request by mitchfaulcon**
### Purpose

Implemented the frontend for the charts page of the dashboard

[HRT-74](https://softeng761team5.atlassian.net/browse/HRT-74)

### Approach

- Added a similar page to the main dashboard page to view the charts for a specific course
- Pages are accessed via /Course/{courseID}
- Added more dummy data for temporary use, currently only the first course has associated charts
- Charts can be selected and the selected charts are stored in an array so that they can later be compared

***FYI: The Compare Charts text is currently off the button but will look fine once #37  is merged in***

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

#### Does this PR involve a UI change?

- [x] Yes
- [ ] No

#### Screenshot(s) of UI before this PR

n/a
#### Screenshot(s) of UI after this PR

image

### CODE REVIEW

**Review by gargi-bansal — CHANGES_REQUESTED** [review]
Mostly looks good to me, just a few UI changes.
- In one of my PRs I changed the styling of the plus image because it's a bit too big (I think in the preview PR). Probably should make it consistent.
- The checkboxes should be a different colour so that they're a bit more visible and obvious.
- The compare button could probably be disabled until at least 2 charts are selected.

File: src/components/Chart.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+interface ChartProps {
+  createdAt: Date;
+  chartID: string;
+  isSelected: boolean;
+  onChange: Function;
+}
+
+const Chart: React.FC<ChartProps> = (props: ChartProps) => {
+  const [chartProps, setChartProps] = useState(props);
```
Comment (gargi-bansal) [inline src/components/Chart.tsx]: is it necessary to store chart props? I assume the only thing you really need isSelected as state? The rest can just be accessible through props

**Comment by mitchfaulcon** [comment]
> * In one of my PRs I changed the styling of the plus image because it's a bit too big (I think in the preview PR). Probably should make it consistent.

Yeah I saw that, I updated the classname for the new one I added so it should look fine once both our PR's are merged in

File: src/components/Course.tsx [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
 }
 
-const Dimension: React.FC<CourseProps> = (props: CourseProps) => {
+const Course: React.FC<CourseProps> = (props: CourseProps) => {
+  let history = useHistory();
+
   const onClick = () => {
-    // TODO: Redirect to appropriate course page
+    history.push("/Course/" + props.courseID);
   };
```
Comment (gargi-bansal) [inline src/components/Course.tsx:17]: Not necessary to change but you can also use backticks to put variables in a string.
E.g. history.push(`/Course/${props.courseID}`)

**Review by gargi-bansal — APPROVED** [review]

**Review by dder725 — APPROVED** [review]
LGTM!

**Pull request merged**